README.md

```diff
@@ -121,15 +121,75 @@ but it doesn't traverse the object recursively converting nested items:
 ```
 
 
-## Public API
+## API
+
+### Extending `itemadapter`
+
+This package allows to handle arbitrary item classes, by implementing an adapter interface:
+
+_class `itemadapter.adapter.AdapterInterface(item: Any)`_
+
+Abstract Base Class for adapters. An adapter that handles a specific type of item should
+inherit from this class and implement the abstract methods defined on it. `AdapterInterface`
+inherits from [`collections.abc.MutableMapping`](https://docs.python.org/3/library/collections.abc.html#collections.abc.MutableMapping),
+so all methods from the `MutableMapping` class should be implemented as well.
```

Comment:
Instead of all of these methods being abstract, wouldn’t it make sense to provide some default implementations? e.g. a `get_field_meta` that returns empty metadata, a `field_names` method that simply returns the fields from the `MutableMapping` interface, and an `asdict` that assumes that `dict(item)` works for root-level conversion.